Change project LICENSE to MIT

wdlkmpx opened the pull request:
to avoid incompatibility with the licenses used by the OpenSSL project

https://en.wikipedia.org/wiki/MIT_License

wdlkmpx commented:
ref #130 

Deleted relicensed code...
[gftp__dont_use_proxy.zip](https://github.com/masneyb/gftp/files/7585635/gftp__dont_use_proxy.zip)

Files with other (c) holders that must agree with the license change...
- lib/fsp.c - @hsn10
- src/gtk/platform_specific.c @sedwards 

And someone who probably provided the most important fix:
- @RocketMan 

Most other important contributions have been blurred or removed by my crazy changes, so I guess that's it...

masneyb commented on lib/cache.c:
```diff
@@ -1,21 +1,25 @@
-/*****************************************************************************/
+/***********************************************************************************/
+/*  cache.c - contains the cache routines                                          */
+/*  Copyright (C) 1998-2021 Brian Masney <masneyb@gftp.org>                        */
```
Per the issue, I'm ok with this for my contributions. This needs one minor change though. The old line had 1998-2007 for my changes to this file, but this changed to include 2021. I think you should leave my copyright lines alone since I haven't modified these files in years. You can add a new line if you'd like:

```
Copyright (C) 1998-2007 Brian Masney <masneyb@gftp.org> 
Copyright (C) 2020-2021 YOUR NAME <...>
```

masneyb commented on LICENSE:
```diff
@@ -0,0 +1,19 @@
+Copyright (c) 1998-2021 Brian Masney
```
Per my comment below, you should also add a second line copyright line here for you as well and throughout if you desire.

hsn10 commented:
I have no problems with MIT license

gusnan commented:
I have only tiny contributions myself, but I think I might as well for completeness claim that I have no problems with changing license on my contributions (if any).

And also chime in and give a big thank you for making this (Handling licensing properly is the really boring part of free software development).

Thanks to all present and previous developers!

RocketMan commented:
You have my support to make the license change.

@wdlkmpx thank you for this, and for all your contributions!!

wdlkmpx commented:
I fixed the (C) years, I also added myself to a couple C files ... and the main LICENSE file

https://github.com/masneyb/gftp/compare/e685f7ef91690bd778fa7a140af4a86f41936a8e..0d410c05829da7fc64fe4470de133942662bac37

Revised relicensed deleted code:
[gftp__dont_use_proxy-MIT-2.c.zip](https://github.com/masneyb/gftp/files/7590259/gftp__dont_use_proxy-MIT-2.c.zip)

I'll try to restore `FTP over HTTP Proxy` support, change license to MIT, and then potentially delete it again haha
https://github.com/masneyb/gftp/commit/9a5b095b2417880f7d972f2b0fefbc5732839676

I say this because the Proxy stuff is such an obscure topic, I had a hard time trying to figure out how to use gFTP with ftp.proxy (http://ftpproxy.org/), with HTTP proxy servers it becomes even more confusing, is it even possible now that HTTP has become a giant monster and FTP support has been removed from the main browsers? That is the question

masneyb commented:
Looks good to me!

wdlkmpx commented:
Ok everyone agrees with the change, that's great. It will happen soon.

We'll wait a few days for @sedwards explicit approval of this change (it's a given)
- src/gtk/platform_specific.c  (macOS-specific code)

wdlkmpx commented:
I renamed 3 files, I also retored the broken HTTP/HTTPS support. So I had to rebase

- lib/protocol_fsp.c
- lib/protocol_ftp.h
- lib/protocol_localfs.c

- lib/protocol_http.c
- lib/protocol_http.h
- lib/protocol_https.c
